src/interchange/src/avro.rs

```diff
@@ -2225,21 +2212,36 @@ pub fn column_names_and_types(desc: RelationDesc) -> Vec<(ColumnName, ColumnType
     columns
 }
 
-/// Encodes a sequence of `Datum` as Avro, using supplied column names and types.
-pub fn encode_datums_as_avro<'a, I>(datums: I, names_types: &[(ColumnName, ColumnType)]) -> Value
+/// Encodes a sequence of `Datum` as Avro (key and value), using supplied column names and types.
+pub fn encode_datums_as_avro<'a, I>(
```

Comment:
nit / totally feel free to disregard - I think I would have constructed the `value_fields` array as before, and then at the end if there was a `key_filter` I would have grabbed stuff from `value_fields` to stuff into `key_fields` I think thats arguably less efficient but its a lot simpler code.

That being said, I admit this is 100% personal preference so feel free to tioli :)

Reply:
actually the way I did it isn't even correct, because it only works if the keys are specified in the same order as they appear in the input.